src/game/AI/ScriptDevAI/scripts/outland/netherstorm.cpp

```diff
@@ -1580,6 +1580,232 @@ CreatureAI* GetAI_npc_dimensius(Creature* pCreature)
+
+    uint32 m_uiAttackTimer;
+    bool m_uiFoundBalls;
+    uint32 m_uiGravityFlux;
+    bool m_uiUsedMirrorImage;
+    uint32 m_uiStasisTimer;
+    std::vector<ObjectGuid> m_uiSummoned;
+
+    void Reset() override
+    {
+        m_uiAttackTimer = 0;
+        m_uiGravityFlux = 15000;
+        m_uiUsedMirrorImage = true;
+        m_uiStasisTimer = 10000;
+        m_creature->RemoveAurasDueToSpell(SPELL_SALAADIN_OVERSPARK);
+        m_creature->CastSpell(m_creature, SPELL_FLOAT, TRIGGERED_NONE);
+        for (const auto& summonedCreature : m_uiSummoned)    // despawn all summoned creatures
+            if (Creature* summon = m_creature->GetMap()->GetCreature(summonedCreature))
+                summon->ForcedDespawn(0);
+    }
+
+    void ReceiveAIEvent(AIEventType eventType, Creature* /*pSender*/, Unit* /*pInvoker*/, uint32 /*uiMiscValue*/) override
+    {
+        if (eventType == AI_EVENT_CUSTOM_A && (!m_creature->isInCombat()) && (!m_creature->isDead()))
+        {
+            m_creature->RemoveAurasDueToSpell(SPELL_FLOAT);
+            DoScriptText(SAY_THREAT, m_creature);
+            m_uiAttackTimer = 5000;
+        }
+    }
+
+    void JustDied(Unit* killer) override
+    {
+        for (const auto& summonedCreature : m_uiSummoned)    // despawn all summoned creatures
+            if (Creature* summon = m_creature->GetMap()->GetCreature(summonedCreature))
+                summon->ForcedDespawn(0);
+    }
+
+    void JustReachedHome() override
+    {
+        m_creature->ForcedDespawn(); // despawn at return home
+    }
+
+    void JustRespawned() override
+    {
+        m_uiFoundBalls = false;
+    }
+
+    void JustSummoned(Creature* pSummoned) override
+    {
+        m_uiSummoned.push_back(pSummoned->GetObjectGuid());
+        pSummoned->CastSpell(pSummoned, SPELL_SALAADIN_OVERSPARK, TRIGGERED_OLD_TRIGGERED);
+        pSummoned->AI()->AttackStart(m_creature->getVictim());
+    }
+
+    void UpdateAI(const uint32 uiDiff) override
+    {
+        if (m_uiFoundBalls == false)
+        {
+            std::list<Creature*> creatureList;
+            GetCreatureListWithEntryInGrid(creatureList, m_creature, NPC_ENERGY_BALL, 50.0f);
+            for (std::list<Creature*>::const_iterator itr = creatureList.begin(); itr != creatureList.end(); ++itr)
+            {
+                m_uiFoundBalls = true;
+                m_creature->AI()->SendAIEvent(AI_EVENT_CUSTOM_A, m_creature, (*itr));
+            }
+            m_creature->SetSheath(SHEATH_STATE_UNARMED);
+        }
+        if (m_uiAttackTimer)
+        {
+            if (m_uiAttackTimer <= uiDiff) // on attack start cast spell that handles everything
+            {
+                m_uiAttackTimer = 0;
+                std::list<Player*> playerList;
+                GetPlayerListWithEntryInWorld(playerList, m_creature, 100.0f);
+                std::list<Player*>::const_iterator itr = playerList.begin();
+                if (itr != playerList.end())
```

Comment:
Let's say it's not a big deal either way, but it was a good excuse to show I'm still alive :wink: